README.md

```diff
@@ -57,35 +54,42 @@ time, `imagePullPolicy: Always` is needed.
 Release process
 ---------------
 
-* Maintainers tag individual components.
+* A change is submitted against master.
+
+* A [Prow](https://github.com/kubernetes/test-infra/blob/master/prow/README.md)
+  job checks out the source of a modified component, rebuilds it and then
+  runs unit tests and E2E tests with it.
 
-* A single
-  [Prow](https://github.com/kubernetes/test-infra/blob/master/prow/README.md)
-  job checks out the source of all components and builds them.
+* Maintainers accept changes into the master branch.
 
-* Container images are placed in a staging area specific to the current build.
-  Staging areas from previous builds that are older than a certain amount
-  of time are garbage collected to free up the space.
+* The same Prow job runs for master and repeats the check. If it succeeds,
+  a new "canary" image is published for the component.
 
-* The job runs component unit tests (aka `make test` in each component
-  repo), sanity and E2E tests (from
-  [csi-test](https://github.com/kubernetes-csi/csi-test), with the
-  definition of what to test in which configurations stored in this
-  repo).
+* In preparation for the release of a major new update, a feature freeze is
+  declared for the "master" branch and only changes relevant for that next
+  release are accepted.
 
-* After reviewing results and potentially some more manual tests,
-  someone must trigger the promotion of release artifacts from the
-  staging area to the production area and announce the release.
+* When all changes targeted for the release are in master, automatic
+  test results are okay and and potentially some more manual tests,
+  maintainers tag each component directly on the master branch.
 
+* Maintenance releases are prepared by branching a "release-X.Y" branch from
+  release "vX.Y" and backporting relevant fixes from master. The same
+  prow job as for master also handles the maintenance branches, but potentially
+  with a different configuration.
```

Comment:
My main concern with this approach compared to the previous, centralized proposal is that there are no safeguards in place: if a maintainer makes a mistake during tagging and that mistake doesn't cause a test failure, then the CI will automatically push images into the same place from where people pull their production images.

Reply:
This is consistent with a "release often, roll back on failure" approach of course. I'm just uneasy with that approach because of my background in embedded, where rollback isn't easy and errors can be fatal. :man_shrugging: